site/content/dev-guide/how-tos/implement-an-adapter.md

```diff
@@ -0,0 +1,112 @@
++++
+title="Implement an Adapter"
+weight=56
++++
+
+NoSqlBench is composed of a sound architectural underpinning that makes the development and addition of new adapters as seamless and quick a process as possible. nb provides a hierarchy of classes which can be extended and interfaces to be implemented, in order to avoid having to re-implement the application infrastructure and in so doing reduces both the complexity and demands on time involved in implementing a new adapter type. However, it is necessary to understand this existing architecture in order to effectively use the provided building blocks. The following is intended as a guide to developing new adapters to extend the functionality of nb5.
+
+# Prerequisites and Assumptions
+* It is assumed that the developer(s) have familiarized themselves with the  [NoSqlBench core concepts](https://docs.nosqlbench.io/introduction/core-concepts/) provided in the nosqlbench documentation.
+* The ability to code in Java is required. NoSqlBench makes extensive use of lambda functionality. It is a requirement to understand and be comfortable using these principles to develop an adapter.
+* This guide is not specific to any particular client. It is assumed that the developer(s) have familiarity with the client they are developing an adapter for.
+
+# Internal Dependencies
+The nb project as a whole is composed of numerous modules and packages. For the purpose of customer adapter implementation there are only two required; adapters-api and nb-annotations.
+
+	<dependency>
+		<groupId>io.nosqlbench</groupId>
+     	<artifactId>adapters-api</artifactId>
+     	<version>${revision}</version>
+     	<scope>compile</scope>
+	</dependency>
+		
+	<dependency>
+        <groupId>io.nosqlbench</groupId>
+        <artifactId>nb-annotations</artifactId>
+        <version>${revision}</version>
+        <scope>compile</scope>
+	</dependency>
+	
+The adapters-api and nb-annotations  modules contain all of the base classes, interfaces and annotations required to implement a new adapter and flesh out its functionality.
+
+* Adapters - Every new adapter will, of course, include an Adapter class. This class will be a descendent of the BaseDriverAdapter class found here. Nosqlbench uses the Java Service Provider Interface(SPI) to define service providers. If you don’t know what that is, don't worry about it now. The tl;dr is that extending the BaseDriverAdapter and annotating your driver adapter class with the Service annotation will result in that class being added to META-INF/services under the specified class name, making it available as a service provider. Nosqlbench takes care of all the rest!
```

Comment:
Referenced the BaseDriverAdapter class as found here.  There needs to be a link included.